examples/grid2grid/evaUtilities.py

```diff
@@ -0,0 +1,132 @@
+import matplotlib.pyplot as plt
+from config.config_manager import load_use_case_config
+import numpy as np
+from mpl_toolkits.mplot3d import proj3d
+from matplotlib.patches import Rectangle, Circle, FancyBboxPatch, FancyArrowPatch
+import mpl_toolkits.mplot3d.art3d as art3d
+import mpl_toolkits.mplot3d as mp3d
+import matplotlib.patches as mpatches
+
+
+class Arrow3D(FancyArrowPatch):
+    def __init__(self, xs, ys, zs, *args, **kwargs):
+        FancyArrowPatch.__init__(self, (0,0), (0,0), *args, **kwargs)
+        self._verts3d = xs, ys, zs
+
+    def draw(self, renderer):
+        xs3d, ys3d, zs3d = self._verts3d
+        xs, ys, zs = proj3d.proj_transform(xs3d, ys3d, zs3d, renderer.M)
+        self.set_positions((xs[0], ys[0]), (xs[1], ys[1]))
+        FancyArrowPatch.draw(self, renderer)
+
+
+def wrap_to_pi(angle):
+    # Wrap to 360
+    ang_2pi = angle - (angle // (2 * np.pi)) * 2*np.pi
+    if ang_2pi > np.pi or ang_2pi < - np.pi:
+        # Wrap to 180
+        ang_pi = ang_2pi-np.sign(ang_2pi)*2*np.pi
+        return ang_pi
+
+
+def quaternion_multiply(quaternion1, quaternion0):
+    w0, x0, y0, z0 = quaternion0
+    w1, x1, y1, z1 = quaternion1
+    return np.array([-x1 * x0 - y1 * y0 - z1 * z0 + w1 * w0,
+                     x1 * w0 + y1 * z0 - z1 * y0 + w1 * x0,
+                     -x1 * z0 + y1 * w0 + z1 * x0 + w1 * y0,
+                     x1 * y0 - y1 * x0 + z1 * w0 + w1 * z0], dtype=np.float64)
+
+
+def solve_ik(_eva, guess, theta, xyz_absolute):
+    with _eva.lock():
+        # Inputs: Theta [deg], xyz_absolute [m]
+        pos = [xyz_absolute[0], xyz_absolute[1], xyz_absolute[2]]  # [m]
+        pos_json = {'x': (pos[0]), 'y': (pos[1]), 'z': (pos[2])}
+        # This provides a rotation of theta [deg] wrt the end effector orientation pointing downwards
+        orient_rel = [np.cos(np.deg2rad(theta)/2), 0, 0, np.sin(np.deg2rad(theta)/2)]
+        orient_abs = quaternion_multiply([0, 0, 1, 0], orient_rel)
+        orient_json = {'w': (orient_abs[0]), 'x': (orient_abs[1]), 'y': (orient_abs[2]), 'z': (orient_abs[3])}
+        # Compute IK
+        result_ik = _eva.calc_inverse_kinematics(guess, pos_json, orient_json)
+        success_ik = result_ik['ik']['result']
+        joints_ik = result_ik['ik']['joints']
+    return success_ik, joints_ik
+
+
+def solve_fk(eva, joints):
+    with eva.lock():
+        fk_results = eva.calc_forward_kinematics(joints)
+        pos_json = fk_results['position']
+        orient_json = fk_results['orientation']
+        pos = [pos_json['x'], pos_json['y'], pos_json['z']]
+        orient = [orient_json['w'], orient_json['x'], orient_json['y'], orient_json['z']]
+    return pos, orient
+
+
+def plot_grids(grid_x, grid_y, grid_z, ax_all, _grid_iter):

```

Comment:
Might be worth moving the Plot stuff to a separate file as well, to separate concerns

Reply:
grouped it into the EvaGrids class